.github/workflows/nightly-test.yaml

```
@@ -0,0 +1,61 @@
+name: Nightly Latest/Edge Tests
+
+on:
+  # TODO: remove, this is just testing
+  push: 
+    branches: 
+      - KU-91/nightly-runs
+  schedule:
+    - cron: '0 0 * * *'  # Runs every midnight
+
+jobs:
+  test-integration:
+    name: Integration Test ${{ matrix.os }} ${{ matrix.runner == 'Ubuntu_ARM64_4C_16G_01' && 'arm64' || 'amd64' }}
+    strategy:
+      matrix:
+        os: ["ubuntu:20.04", "ubuntu:22.04"]
+        runner: ["ubuntu-20.04", "Ubuntu_ARM64_4C_16G_01"]
+      fail-fast: false # TODO: remove once arm64 works
+
+    runs-on: ${{ matrix.runner }}
+
+    steps:
+      # TODO: remove once done debugging
+      - name: Setup tmate session
+        uses: mxschmitt/action-tmate@v3
+        with:
+          # Print connection details and continue with the job.
+          # Waits at the end of the job for the tmate session to exit.
+          # If no user connects within 10min the connection exits gracefully.
+          detached: true
+          # Only the user who started the workflow can access the tmate session.
+          limit-access-to-actor: true
+        if: ${{ failure() }}
+      - name: Checking out repo
+        uses: actions/checkout@v4
+      - name: Setup Python
+        run: |
+          sudo apt update
+          sudo apt install -y python3 python3-pip
+      - name: Install tox
+        run: |
+          pip3 install tox==4.13
+      - name: Install lxd
+        run: |
+          sudo snap refresh lxd --channel 5.19/stable
+          sudo lxd init --auto
+          sudo usermod --append --groups lxd $USER
+          sg lxd -c 'lxc version'
+      - name: Create build directory
+        run: mkdir -p build
+      - name: Install latest/edge k8s snap
+        run: |
+          cd build
+          snap download k8s --channel=edge --basename k8s
```

i'd LOVE if this channel was based on which branch this was running on

```suggestion
          snap download k8s --channel=${release} --basename k8s
```

---

Maybe compromise on matrix for now? `releases: ["latest/edge", "1.30-classic/beta"]`?